### PR TITLE
boards: stm: nucleo_g4xx: added flash partitions to work with mcuboot

### PR DESCRIPTION
**Pull request by ghost**
Added code partitioning for ST Nucleo G431RB and G474RE boards. 
Defined flash partitions including "mcuboot," "image-0," and "image-1".

### CODE REVIEW

File: boards/st/nucleo_g431rb/nucleo_g431rb.dts
Comment (nordicjm) [inline boards/st/nucleo_g431rb/nucleo_g431rb.dts:178]: slot0 should be larger than slot1 if using swap using move. If you build MCUboot with debug log mode enabled, it will give you a note about this, you can adjust the sizes which should then make the log output vanish and give an optimal slot size

Comment (nordicjm) [reply boards/st/nucleo_g431rb/nucleo_g431rb.dts:178]: No, that is a cmake warning, you need to build mcuboot with log level set to debug, which will show something like:
```
*** Booting Zephyr OS build v3.6.0-2432-gb318f609a058 ***
I: Starting bootloader
D: Non-optimal sector distribution, slot0 has 62 usable sectors (64 assigned) but slot1 has 64 assigned
```
Those boards have not been updated

Comment (dalprahcd) [reply boards/st/nucleo_g431rb/nucleo_g431rb.dts:178]: Can you help me further? I have increased the size of the slot0 image, but now I get a different warning

```
*** Booting Zephyr OS build v3.6.0-2432-gdcef1df8d73c ***
I: Starting bootloader
W: Cannot upgrade: not a compatible amount of sectors
D: slot0 sectors: 23, slot1 sectors: 21, usable slot0 sectors: 20
I: Bootloader chainload address offset: 0x9000
I: Jumping to the first image slot*** Booting Zephyr OS build v3.6.0-2432-gdcef1df8d73c ***
Hello World! nucleo_g431rb/stm32g431xx
```

I am building with the following command if it helps
```
west build -p always -b nucleo_g431rb/stm32g431xx bootloader/mcuboot/boot/zephyr -d build-mcuboot -- -DCONFIG_MCUBOOT_LOG_LEVEL_DBG=y
```

Comment (nordicjm) [reply boards/st/nucleo_g431rb/nucleo_g431rb.dts:178]: It seems that on your device using your configuration, MCUboot needs 3 sectors for swapping and storing update information (this seems high, what have you set `CONFIG_BOOT_MAX_IMG_SECTORS` to?), so you would need to set slot1 to 20 sectors instead of 21

Comment (dalprahcd) [reply boards/st/nucleo_g431rb/nucleo_g431rb.dts]: Thanks for your help, I fixed the flash partitions now the warning is gone:

```
*** Booting Zephyr OS build v3.6.0-2432-gdcef1df8d73c ***
I: Starting bootloader
I: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
W: Failed reading image headers; Image=0
E: Unable to find bootable image
```

**Review by nordicjm — DISMISSED** [review]

**Review by ajarmouni-st — DISMISSED** [review]
Thanks for contributing!

**Comment by dalprahcd** [comment]
Hello @nordicjm and @ajarmouni-st 

I have dismissed your approval because I have added new changes, more specifically, I am defining the `mcuboot-led0` as well. Could you kindly review it again?

**Review by nordicjm — APPROVED** [review]

**Review by ajarmouni-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**
